### ISSUE TITLE
CRD redesign

### ISSUE THREAD

**Issue by razvan** [stackabletech/spark-k8s-operator#273]
# Description

As discussed [here](https://github.com/stackabletech/spark-k8s-operator/pull/272#pullrequestreview-1583916385), the CRD of Spark applications should become more consistent with CRDs used by other Stackable operators.

```[tasklist]
### Tasks
- [ ] Move jvmSecurity and podOverrides to a common configuration structure.
```

**Comment by fhennig** [comment]
The Spark history server does not expose a "role level" service, only role group level services. Also, the listernerClass setting affects the role group service (it usually doesn't do that for other products).

I would suggest to introduce a role level service, which is then also using the listenerClass. The role group level services should always be ClusterIP.

**Comment by lfrancke** [comment]
I assume this is a breaking change in CRD?

**Comment by razvan** [comment]
Yes. as discussed  [here](https://github.com/stackabletech/spark-k8s-operator/pull/275#issuecomment-1695934727).
